auth.py

```diff
@@ -43,22 +43,22 @@ def extract_url(request_uri):
     return None
 
 
-def get_entry(url):
-    for k, v in acl.items():
+def get_entry(url, port):
+    for k, v in acl[port].items():
         if url.startswith(k):
             return v
     return None
```

Comment:
ポートが定義されていない場合、 `KeyError` になって落ちます。
該当するポートがなければアクセス制限を課すべきではないため、後続の処理が期待通りに動作するように空の辞書を返すのはいかがでしょうか。
また、 #1 で見落としましたが、URLが見つからなかった時に返す値も同様です。

```suggestion
def get_entry(url, port):
    if port not in acl:
        return {}
    for k, v in acl[port].items():
        if url.startswith(k):
            return v
    return {}
```
※動作確認は未実施

Reply:
https://github.com/udzuki/spnego-nginx-auth-extension/blob/main/auth.py#L56

元々の実装では、entryが無かったらNoneになって、if entryでチェックして{}にしてるので、返すとしたら{}ではなくNoneかと思いましたが、如何でしょう？

Reply:
あ、その通りですね。
返す値は `None` のままで良いです。勘違いしてました。
それでは以下だけ追加すれば良いかと思います。

```python
    if port not in acl:
        return None
```

Reply:
追加しました。